### PR TITLE
Process source node CiliumEndpoints in CEB watch events

### PR DESCRIPTION
**Pull request by krishgobinath**
Update CEPs in all cilium nodes including source node CEPs.

Found couple of issues while running CI tests against CEB feature enabled default.

1) If CEP's are updated through CEB watch events, process CEPs in all nodes. Currently, we are
skipping CEP updates in source nodes. 
Here is why we need to update on all nodes...
a) If a pod is deleted, and the corresponding CEP is removed from source node. If CEPs are updated through CEB watcher, in `OnDelete` [function](https://github.com/cilium/cilium/blob/feature/cep-scalability/pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go#L136-L150) we are unable to identify the CEP is local to the node or not, in the below case `LookupPodName` returns `nil` for source node CEP, hence we are calling `cs.kWatcher.endpointDeleted`

```
    if p := cs.kWatcher.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(c)); p != nil {
			continue
   }
```
However, in `endpointDeleted`, we were trying to [delete ipCache](https://github.com/cilium/cilium/blob/feature/cep-scalability/pkg/k8s/watchers/cilium_endpoint.go#L230-L235) but that ends up in cilium-agent crash because for source node CEP's we never upserted `ipcache` as mentioned [here](https://github.com/cilium/cilium/blob/feature/cep-scalability/pkg/k8s/watchers/cilium_endpoint.go#L178-L212) because we always skip CEP updating for source node CEPs.

2) Program QPS limit and Burst limit to default values

In our code default qps and burst limit values are zero `option.Config.K8sClientQPSLimit` and `option.Config.K8sClientBurst`, unless they are configured in cilium configmap.

If the given `qps, burst` values are zero k8s client will override with defaults values, as mentioned [here](https://github.com/cilium/cilium/blob/master/pkg/defaults/defaults.go#L359-L365). 

For workqueues, if `qps limit` is zero, then 'queue.AddRateLimited` API isn't working, it's not adding the `work item` back into queue. Hence we need to program default qps, burst  values for workqueues.

Signed-off-by: Gobinath Krishnamoorthy gobinathk@google.com

### CODE REVIEW

File: operator/pkg/ciliumendpointbatch/endpointbatch.go [hunk 1]
```diff
@@ -52,6 +52,10 @@ const (
 	maxRetries = 15
 	// CEPs are batched into a CEB, based on its Identity
 	cebIdentityBasedBatching = "cebBatchModeIdentity"
+	// default qps limit value for workqueues, this only for retries.
+	workQueuesQPSLimit = 10
+	// default burst limit value for workqueues.
+	workQueuesBurstLimit = 100
```
Comment (Weil0ng) [inline operator/pkg/ciliumendpointbatch/endpointbatch.go]: is this too high? 100 qps for CEB updates seems like it could easily kill apiserver?

Comment (krishgobinath) [reply operator/pkg/ciliumendpointbatch/endpointbatch.go]: QPS is 10, but burst is 100. These values are inline with endpoint slice implementation. Irrespective of that K8S client would cap the number of requests going into api-server.

Comment (Weil0ng) [reply operator/pkg/ciliumendpointbatch/endpointbatch.go]: sgtm, another small nit is that these two default values should have more specific names, workqeuue is too general, some other component might also use workqueue and want a different QPS/Burst.

File: pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go [hunk 2]
```diff
@@ -49,7 +49,6 @@ func (cs *cebSubscriber) OnAdd(ceb *cilium_v2a1.CiliumEndpointBatch) {
 		if p := cs.kWatcher.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(c)); p != nil {
```
Comment (aanm) [inline pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go:47]: The comment on line 47-48 can be removed now.

File: pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go [hunk 3]
```diff
@@ -103,7 +102,6 @@ func (cs *cebSubscriber) OnUpdate(oldCEB, newCEB *cilium_v2a1.CiliumEndpointBatc
 			if p := cs.kWatcher.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(c)); p != nil {
```
Comment (aanm) [inline pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go:98]: The comment on line 100-101 can be removed now.

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by Weil0ng — APPROVED** [review]

File: pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go (hunk 2 above)
Comment (krishgobinath) [reply pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go:47]: Done.

File: pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go (hunk 3 above)
Comment (krishgobinath) [reply pkg/k8s/watchers/cilium_endpoint_batch_subscriber.go:98]: Done.

**Review by aanm — APPROVED** [review]

**Pull request merged**